### PR TITLE
[consensus] replace direct send with reliable broadcast in buffer man…

### PR DESCRIPTION
**Pull request by zekun000**
…ager

### CODE REVIEW

**Review by sasha8 — APPROVED** [review]
Fix the bug :)

File: consensus/src/network.rs [hunk 1]
```diff
@@ -509,19 +542,14 @@ impl NetworkTask {
...
-        );
         let (quorum_store_messages_tx, quorum_store_messages) = aptos_channel::new(
             QueueStyle::FIFO,
             // TODO: tune this value based on quorum store messages with backpressure
             50,
             Some(&counters::QUORUM_STORE_CHANNEL_MSGS),
         );
         let (rpc_tx, rpc_rx) =
-            aptos_channel::new(QueueStyle::LIFO, 1, Some(&counters::RPC_CHANNEL_MSGS));
+            aptos_channel::new(QueueStyle::FIFO, 10, Some(&counters::RPC_CHANNEL_MSGS));
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/network.rs:552]: because it may have multiple RPC in fly now

[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/network.rs:552]: it's just random numbers :)

File: consensus/src/network.rs [hunk 2]
```diff
@@ -207,6 +216,35 @@ impl NetworkSender {
         Ok(response)
     }
 
+    pub async fn send_rpc(
```
Comment (brianolson) [inline consensus/src/network.rs:219]: might need to adjust comment on line 155

File: consensus/src/experimental/mod.rs [hunk 3]
```diff
@@ -32,6 +32,7 @@ pub mod hashable;
 pub mod ordering_state_computer;
 pub mod persisting_phase;
 pub mod pipeline_phase;
+pub mod reliable_broadcast;
```
Comment (ibalajiarun) [inline consensus/src/experimental/mod.rs:35]:
```suggestion
pub mod commit_reliable_broadcast;
```

File: consensus/src/experimental/buffer_manager.rs [hunk 4]
```diff
@@ -413,24 +450,30 @@ impl BufferManager {
...
+                        if let Err(e) = sender.send_commit_vote(commit_vote, proposer).await {
+                            warn!("Failed to send commit vote {:?}", e);
+                        }
+                    });
                 } else {
-                    self.commit_msg_tx.broadcast_commit_vote(commit_vote).await;
+                    let commit_vote = CommitMessage::Vote(commit_vote);
+                    signed_item_mut
+                        .rb_handle
+                        .replace((Instant::now(), self.do_reliable_broacast(commit_vote)));
```
Comment (ibalajiarun) [inline consensus/src/experimental/buffer_manager.rs:474]: The timestamp seems unused?

File: consensus/src/experimental/buffer_manager.rs [hunk 5]
```diff
@@ -152,6 +177,16 @@ impl BufferManager {
         }
     }
 
+    fn do_reliable_broacast(&self, message: CommitMessage) -> DropGuard {
```
Comment (ibalajiarun) [inline consensus/src/experimental/buffer_manager.rs:180]:
```suggestion
    fn do_reliable_broadcast(&self, message: CommitMessage) -> DropGuard {
```

File: consensus/src/experimental/buffer_manager.rs (hunk 4 above)
Comment (zekun000) [reply consensus/src/experimental/buffer_manager.rs:474]: it's used to decide to re-initiate the rb

**Review by ibalajiarun — APPROVED** [review]

**Pull request merged**